examples/text-generation/utils.py

```diff
@@ -515,6 +515,12 @@ def initialize_model(args, logger):
         "token": args.token,
         "trust_remote_code": args.trust_remote_code,
     }
+
+    # For starcoderbase model it is essential to setup proper flags for sdpa kernel at model level
+    # to avoid transpose of tensors at each attention call
+    if 'starcoderbase' in args.model_name_or_path:
```

Comment:
Does this also apply to bigcode/starcoder? If so is it better to check the model type is gpt_bigcode instead?

Reply:
hm, right it might be better approach. But need to plug in to at least two functions: ```setup_model``` and ```setup_distributed_model```

Reply:
ok, there is one problem with this approach. Currently (see [here](https://github.com/huggingface/optimum-habana/blob/main/examples/text-generation/utils.py#L214)) we don't extract ```config``` ahead model creation. I might add new function, which will extract config just to check model type and modify args. It might be usefull for other models in future as well. If you are fine with such approach, I can do corresponding change

Reply:
I think that sounds better. I too ran into issues when I tried to find the right place to insert this, so a refactor might be in order. The main requirement was for it to apply to gpt_bigcode models in general, example: starcoderbase and starcoder - see my comment: https://github.com/huggingface/optimum-habana/pull/1138#pullrequestreview-2184281140

Reply:
Capturing some options we discussed offline:

Another way that I tested is to pass it during initialize_model() in examples/text-generation/utils.py like so
```
    model_kwargs = {
        "revision": args.model_revision,
        "token": args.token,
        "trust_remote_code": args.trust_remote_code,
        "use_flash_attention": args.use_flash_attention,
    }
```
But if you find you can override the classes so flash_attention is passed in early enough and recognized i.e from setup_generation_config() of examples/text-generation/utils.py then that is probably best.